packages/flutter_tools/lib/src/globals.dart

```diff
@@ -13,6 +14,7 @@ Logger get logger => context[Logger];
 Cache get cache => Cache.instance;
 Config get config => Config.instance;
 Artifacts get artifacts => Artifacts.instance;
+ApplicationPackageFactory get applicationPackageFactory => ApplicationPackageFactory.instance;
```

Comment:
Let's just refer to `ApplicationPackageFactory.instance` explicitly where we need it - we're trying to reduce the amount of globals we have.